docs/design/ConstructiveSolidGeometry.MD

```diff
@@ -0,0 +1,136 @@
+# Constructive Solid Geometry
+
+[Constructive Solid Geometry](https://en.wikipedia.org/wiki/Constructive_solid_geometry) (CSG) allows the modeler to create
+complex shapes using boolean operators to combine simpler
+solid objects.
+
+![](https://upload.wikimedia.org/wikipedia/commons/thumb/8/8b/Csg_tree.png/450px-Csg_tree.png)
+
+In Shards we employ a [Signed Distance Function](https://en.wikipedia.org/wiki/Signed_distance_function) (SDF) to generate implicit 3D shapes such as spheres, cubes, capsules, torii etc. which can be traced by ray-marching through the scene.
+
+We'll explore in depth the binary tree representation, how to flatten it to a list, how the shader consumes it and what's the math behind it.
+
+## The Tree
+
+Each node in the tree is either a simple shape (a leaf node) or an operator that takes the result of one or more child nodes (operands) and combines them. For simplicity we consider unary and binary operators only. Operations on three or more elements can be split into binary operations by nesting some of the elements deeper in the tree. In the example above the right node is an union of three cylinders, but it has been split into two binary union operators.
+
+### Operators
+
+`Union` - combines two shapes, A and B, so that each point in the resulting shapes belongs to either A or B, or both. The two shapes do not need to touch. The resulting shape will visually appear to be two shapes, A and B, even though mathematically we treat it as one. In the example above the three green cylinders form an union.
```

Comment:
Remove comma
`either A or B, or both` => `either A or B or both`

Reply:
Same comment as above (article for `union`)
`cylinders form an union` => `cylinders form a union`